src/api/app/models/staging_project.rb

```diff
@@ -97,47 +97,46 @@ def check_state
+
+    # data of each pkg/repo/arch combo with state broken or failed
     @broken_packages = []
+    # data and count of all repos that are currently building
     @building_repositories = []
+
     buildresult.elements('result') do |result|
-      building = false
-      if !['published', 'unpublished'].include?(result['state']) || result['dirty'] == 'true'
-        building = true
-      end
+      building = ['published', 'unpublished'].exclude?(result['state']) || result['dirty'] == 'true'
+
       result.elements('status') do |status|
         code = status.get('code')
-        if ['broken', 'failed'].include?(code) || (code == 'unresolvable' && !building)
-          @broken_packages << { 'package'    => status['package'],
-                                'project'    => name,
-                                'state'      => code,
-                                'details'    => status['details'],
-                                'repository' => result['repository'],
-                                'arch'       => result['arch'] }
+
+        if code.in?(['broken', 'failed']) || (code == 'unresolvable' && !building)
+          @broken_packages << { package:    status['package'],
+                                project:    name,
+                                state:      code,
+                                details:    status['details'],
+                                repository: result['repository'],
+                                arch:       result['arch'] }.with_indifferent_access
         end
       end
+
       if building
-        # determine build summary
         current_repo = result.slice('repository', 'arch', 'code', 'state', 'dirty')
         current_repo[:tobuild] = 0
         current_repo[:final] = 0
-
-        buildresult = Buildresult.find_hashed(project: name, view: 'summary', repository: current_repo['repository'], arch: current_repo['arch'])
-        buildresult = buildresult.get('result').get('summary')
-        buildresult.elements('statuscount') do |sc|
-          if ['excluded', 'broken', 'failed', 'unresolvable', 'succeeded', 'excluded', 'disabled'].include?(sc['code'])
-            current_repo[:final] += sc['count'].to_i
+        result.get('summary').elements('statuscount') do |status_count|
+          if status_count['code'].in?(['excluded', 'broken', 'failed', 'unresolvable', 'succeeded', 'excluded', 'disabled'])
+            current_repo[:final] += status_count['count'].to_i
           else
-            current_repo[:tobuild] += sc['count'].to_i
+            current_repo[:tobuild] += status_count['count'].to_i
           end
         end
+
         @building_repositories << current_repo
       end
     end
-    if @building_repositories.present?
-      @broken_packages = @broken_packages.reject { |p| p['state'] == 'unresolvable' }
-    end
+
+    @broken_packages.reject! { |p| p['state'] == 'unresolvable' } if @building_repositories.present?
```

Comment:
Better not to still use `p` as a variable name

Reply:
I am surprised one single comment was enough... :unamused: